pkg/controller/storagecluster/placement.go

```diff
@@ -33,13 +33,6 @@ func getPlacement(sc *ocsv1.StorageCluster, component string) rookv1.Placement {
 		}
 	}
 
-	topologyMap := sc.Status.NodeTopologies
-	if topologyMap != nil && (component == "mon" || component == "mds") {
-		topologyKey := determineFailureDomain(sc)
-		topologyKey, _ = topologyMap.GetKeyValues(topologyKey)
-		podAffinityTerms := placement.PodAntiAffinity.PreferredDuringSchedulingIgnoredDuringExecution
-		podAffinityTerms[0].PodAffinityTerm.TopologyKey = topologyKey
-	}
```

Comment:
I have had this thought since a long time now. Good that you brought it up.FYI :  @rohantmp